### ISSUE TITLE
zoom behaviour not as indicated

### ISSUE THREAD

**Issue by splace** [mate-desktop/eom#157]
#### Expected behaviour

image zooms to fit window when that button pressed

#### Actual behaviour

zooms to original size

#### Steps to reproduce the behaviour

view image smaller than window

Eye of MATE 1.12.2
Linux Distribution

Ubuntu mate 16.04

**Comment by raveit65** [comment]
Are you speaking about pressing `F` ?
Pressing `F' fits the image always to current window size here and not to original image size
With eom-1.20.x.

**Comment by splace** [comment]
>  Are you speaking about pressing F ?

yes, well the 'fit' button

>  Pressing F fits the image always to current window size here and not to original image size

not here, the image is zoomed to original size, 100%, that is smaller and not fitted to the window, which would need to be over 100% for smaller sized images. 

BTW i no longer have EOM installed, this, and the other bug i reported,  were enough for me to look for another solution.

**Comment by raveit65** [comment]
Well, thanks for reporting bugs for outdated versions.

**Comment by splace** [comment]
so this is fixed then?

**Comment by splace** [comment]
i'm running solus 3, it has an 'image viewer' app, version 3.28.2, whose help page redirects to EOM.

it has only a 'f' shortcut key  and not a 'fit' button.

so i would guess this has the same code base, but a fairly new version, which is confirmed by it having the same bug.

are you reading my report accurately and trying images whose 100% is smaller than the window being fit too?

**Comment by raveit65** [comment]
for the records.....

**Comment by splace** [comment]
yes, i think you have it, if the image is smaller than the window, fit doesn't work. so you will confirm?

not sure what you mean about "UHD"? or "normal window size"?

bear in mind, screens are getting higher and higher resolution, so more and more images have less pixels than the window you are trying to view them in.

if "fit" were called "shrink to fit" then maybe it would be acceptable, but why? this makes little sense, increase image size works past 100%, obviously.

**Comment by raveit65** [comment]
UHD = Ultra HD resolutions of an image (3840x3160px)
So, 100% level of this image will always bigger than **normal** window size of  eom.
Understand?
I think image resolutions getting higher too, not only monitors. 
In my test with such a huge image , the image will always fit to window size. Doesn't matter if zoom level is higher or lower than window size.
Only small images, eg. 200x200px, doesn't fit to window size.
So this is more a feature request for me.

**Comment by splace** [comment]
this is about images smaller than the window, the absolute size is not relevant.

this is simpler than you seem to think.

> I think image resolutions getting higher too, not only monitors.

having bigger images does not mean small ones go away.

**Comment by raveit65** [comment]
> i'm running solus 3, it has an 'image viewer' app, version 3.28.2, whose help page redirects to EOM.

I am pretty sure you mean eog (eye of gnome), eom doesn't have such a version.
https://download.gnome.org/sources/eog/3.28/
Please file out a report against eog (gnome bugzila) https://bugzilla.gnome.org/
or solus 3

**Comment by splace** [comment]
> >  i'm running solus 3, it has an 'image viewer' app, version 3.28.2, whose help page redirects to EOM.

>I am pretty sure you mean eog (eye of gnome)

bad guess, please re-read.
